### ISSUE TITLE
SyntaxError: Octal literal in strict mode

### ISSUE THREAD

**Issue by archcorsair** [checkly/puppeteer-recorder#66]
Puppeteer-recorder is grabbing this selector:
```js
await page.waitForSelector('.error #\35 11\3A 0')
```
causing Puppeteer to throw this error: `SyntaxError: /Users/nxc/w/salesforce-puppeteer/test/recorded.test.js: Octal literal in strict mode (73:39)`

I suspect this might be a unique issue to the product I'm working with because it is a salesforce based application where it renders nested iFrames.
Creating a test script manually works, using a helper I wrote to fetch the nested frame I need:
```js
const getChildFrame = (parent, index) => {
  const wrapperFrame = parent.mainFrame().childFrames()[index]
  return wrapperFrame.childFrames()[0]
}
```

Any suggestions regarding this specific use case would be appreciated

**Comment by archcorsair** [comment]
Additional screenshot for context:
![image](https://user-images.githubusercontent.com/1036921/65641408-7d835b00-dfa1-11e9-8537-ea7b62f0c30e.png)

**Comment by archcorsair** [comment]
**Update:**
Doing some personal investigation lead me to these two resources:

- https://mathiasbynens.be/notes/html5-id-class

- https://stackoverflow.com/questions/5672903/can-i-have-a-div-with-id-as-number/5672936#5672936

I've discovered the issue is that there is a limitation in CSS itself where if the id of the element starts with a number, it must be escaped. This is causing issues because the escape character itself `\` is not being escaped when generating the script. I was able to fix the above error by additionally escaping the escape character.

Example: (was)
```js
await page.click('.error #\35 11\3A 0')
```
Now:
```js
await page.click('.error #\\35 11\\3A 0')
```
